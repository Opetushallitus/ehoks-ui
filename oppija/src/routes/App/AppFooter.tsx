import { Link } from "@reach/router"
import React from "react"
import { FormattedMessage, intlShape } from "react-intl"
import styled from "styled"
import ehoksLogo from "./ehoks_logo_2.png"
import ophLogo from "./oph_logo.svg"

const EHOKSLogo = styled("img")`
  height: 64px;
  margin: 0 0 40px 0;

  @media screen and (max-width: ${props => props.theme.breakpoints.Tablet}px) {
    flex-direction: row;
    margin: 0 20px 0 0;
  }
`

const OPHLogo = styled("img")`
  height: 50px;
`

const FooterContainer = styled("footer")`
  width: 100%;
  background-color: #fff;
  border-top: 1px solid #d5d8da;
`

const Footer = styled("div")`
  max-width: ${props => props.theme.maxWidth}px;
  margin: 30px auto 0 auto;
  display: flex;
  flex-direction: column;
`

const TopContent = styled("div")`
  display: flex;
  @media screen and (max-width: ${props => props.theme.breakpoints.Tablet}px) {
    flex-direction: column;
  }
`

const FooterColumn = styled("div")`
  display: flex;
  flex: 1;
  justify-content: center;
  flex-direction: column;
`

const Description = styled(FooterColumn)`
  align-items: flex-end;
  border-top: 1px solid #eaeaea;
  border-bottom: 1px solid #eaeaea;
  margin: 20px 0 20px 40px;

  @media screen and (max-width: ${props => props.theme.breakpoints.Desktop}px) {
    margin: 20px 0 20px 20px;
  }

  @media screen and (max-width: ${props => props.theme.breakpoints.Tablet}px) {
    margin: 0 50px 0 20px;
    border-top-width: 0px;
  }
`

const Links = styled(FooterColumn)`
  align-items: center;
  border-top: 1px solid #eaeaea;
  border-bottom: 1px solid #eaeaea;
  margin: 20px 80px 20px 0;

  @media screen and (max-width: ${props => props.theme.breakpoints.Desktop}px) {
    margin: 20px 50px 20px 0;
  }

  @media screen and (max-width: ${props => props.theme.breakpoints.Tablet}px) {
    align-items: flex-start;
    margin: 0 50px 0 20px;
    border-bottom-width: 0px;
  }
`

const DescriptionText = styled("div")`
  width: 70%;
  padding: 12px 0;
  color: #6a6a6a;
  line-height: 1.4;

  @media screen and (max-width: ${props => props.theme.breakpoints.Large}px) {
    width: 90%;
  }

  @media screen and (max-width: ${props => props.theme.breakpoints.Desktop}px) {
    width: 100%;
  }
`

const LinksContainer = styled("div")`
  display: flex;
  flex-direction: column;
  padding: 12px 0;

  a {
    color: #539d1d;
  }
`

const LogosContainer = styled("div")`
  display: flex;
  flex-direction: column;
  padding: 0 70px 0 70px;

  @media screen and (max-width: ${props => props.theme.breakpoints.Tablet}px) {
    flex-direction: row;
    margin: 20px 0;
    align-items: center;
    justify-content: center;
  }
`

const Disclaimer = styled("div")`
  display: flex;
  align-items: center;
  justify-content: center;
  color: #6a6a6a;
  line-height: 1.4;
  margin: 20px 0 20px 0;

  @media screen and (max-width: ${props => props.theme.breakpoints.Desktop}px) {
    margin: 20px 50px 20px 20px;
  }

  @media screen and (max-width: ${props => props.theme.breakpoints.Tablet}px) {
    margin: 20px 50px 20px 20px;
  }
`

export class AppFooter extends React.Component {
  static contextTypes = {
    intl: intlShape
  }

  render() {
    const { intl } = this.context
    return (
      <FooterContainer>
        <Footer>
          <TopContent>
            <Description>
              <DescriptionText>
                <FormattedMessage
                  id="footer.descriptionText"
                  defaultMessage="
              Henkilökohtainen osaamisen kehityssuunnitelma on käytännön
              työväline opiskelijalle, opettajille ja ohjaajille oppilaitoksessa
              ja työpaikoilla."
                />
              </DescriptionText>
            </Description>
            <LogosContainer>
              <EHOKSLogo
                src={ehoksLogo}
                alt={intl.formatMessage({
                  id: "footer.ehoksLogoLabel"
                })}
              />
              <OPHLogo
                src={ophLogo}
                alt={intl.formatMessage({
                  id: "footer.ophLogoLabel"
                })}
              />
            </LogosContainer>
            <Links>
              <LinksContainer>
                <a href="mailto:ehoks@opintopolku.fi">
                  <FormattedMessage
                    id="footer.contactEmailLabel"
                    defaultMessage="Ota yhteyttä: ehoks@opintopolku.fi"
                  />
                </a>
                <Link to="https://beta.opintopolku.fi/konfo/fi/sivu/ehoks-palvelun-tietosuojaseloste">
                  <FormattedMessage
                    id="footer.dataProtectionReportLabel"
                    defaultMessage="Tietosuojaseloste"
                  />
                </Link>
                <Link to="https://beta.opintopolku.fi/konfo/fi/sivu/saavutettavuusselosteet">
                  <FormattedMessage
                    id="footer.accessibilityReportLabel"
                    defaultMessage="Saavutettavuusseloste"
                  />
                </Link>
              </LinksContainer>
            </Links>
          </TopContent>
          <Disclaimer>
            <FormattedMessage
              id="footer.informationDisclaimer"
              defaultMessage="Koulutuksen järjestäjät ylläpitävät eHOKS-tietojasi. Tietojen oikeellisuuden voit tarkistaa kyseisestä oppilaitoksesta."
            />
          </Disclaimer>
        </Footer>
      </FooterContainer>
    )
  }
}
