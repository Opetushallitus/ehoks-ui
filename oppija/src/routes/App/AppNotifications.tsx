import { Link } from "@reach/router"
import { AppContext } from "components/AppContext"
import { Container } from "components/Container"
import { Notification } from "components/Notification"
import { FormattedDate } from "components/FormattedDate"
import { inject, observer } from "mobx-react"
import React from "react"
import { MdClose } from "react-icons/md"
import { FormattedMessage, InjectedIntl, injectIntl } from "react-intl"
import { IAppError } from "stores/ErrorStore"
import { IRootStore } from "stores/RootStore"
import styled from "styled"

const AppNotification = styled(Notification)`
  margin: 0;
`

const Content = styled("div")`
  display: flex;
  flex: 1;
  align-items: center;
`

const Text = styled("div")`
  flex: 1;
  font-size: 16px;
`

const IconContainer = styled("button")`
  display: flex;
  margin: 0 20px;
  cursor: pointer;
  appearance: none;
  padding: 0;
  border: none;
  background: transparent;
`

const NotificationLink = styled(Link)`
  color: ${props => props.theme.colors.waterBlue};
  font-size: 16px;
  margin-right: 10px;
  cursor: pointer;
  text-decoration: underline;
`

const NotificationAnchor = NotificationLink.withComponent("a")

const AlertType = ({ type }: { type: string }) => type === "naytto" ? (
    <FormattedMessage id="muistutukset.naytto" defaultMessage="Näyttö" />
  ) : (
    <FormattedMessage
      id="muistutukset.tyossaoppiminen"
      defaultMessage="Työssäoppiminen"
    />
  )

export interface AppNotificationsProps {
  store?: IRootStore
  intl: InjectedIntl
}

@inject("store")
@observer
export class AppNotifications extends React.Component<AppNotificationsProps> {
  static contextType = AppContext
  declare context: React.ContextType<typeof AppContext>

  ackNotification = (hide: () => void) => (event: React.MouseEvent) => {
    event.preventDefault()
    hide()
  }

  render() {
    const { store, intl } = this.props
    const { featureFlags } = this.context
    const {
      errors: { unhandled },
      notifications
    } = store!
    return (
      <Container>
        {unhandled.map((error: IAppError, i: number) => (
            <AppNotification key={i} type="error">
              <Content>
                <Text>
                  <FormattedMessage id={`errors.${error.id}`} />:{" "}
                  <FormattedMessage
                    id={`errors.${error.errorText}`}
                    defaultMessage={error.errorText}
                  />
                </Text>
                <IconContainer
                  onClick={error.handle}
                  aria-label={intl.formatMessage({
                    id: "errors.piilotaVirheAriaLabel"
                  })}
                >
                  <MdClose size={20} />
                </IconContainer>
              </Content>
            </AppNotification>
          ))}
        {featureFlags.shareNotifications &&
          notifications.visible.map((notification, i) => {
            const message = notification.message
            return (
              <AppNotification
                key={i}
                type="alert"
                iconColor="#000"
                iconSize={32}
              >
                <Content>
                  <Text>
                    <AlertType type={message.type} />, {message.title},{" "}
                    {message.location}{" "}
                    <FormattedDate date={message.startDate} format="d.M." />-
                    <FormattedDate date={message.endDate} />{" "}
                    <FormattedMessage
                      id="muistutukset.onPianAlkamassa"
                      defaultMessage="on pian alkamassa"
                    />
                    .
                  </Text>
                  <NotificationLink
                    to={`/ehoks/suunnittelu/${message.hoksId}/opiskelusuunnitelmani?share=${message.koodiUri}&type=${message.type}`}
                  >
                    <FormattedMessage
                      id="muistutukset.jaaTiedotTyopaikkaohjaajalle"
                      defaultMessage="Jaa tiedot työpaikkaohjaajalle"
                    />
                    .
                  </NotificationLink>
                  <NotificationAnchor
                    role="button"
                    href="#"
                    onClick={this.ackNotification(notification.ackNotification)}
                  >
                    <FormattedMessage
                      id="muistutukset.poistaMuistutus"
                      defaultMessage="Poista muistutus"
                    />
                    .
                  </NotificationAnchor>
                  <IconContainer
                    onClick={notification.hide}
                    aria-label={intl.formatMessage({
                      id: "errors.piilotaVirheAriaLabel"
                    })}
                  >
                    <MdClose size={20} />
                  </IconContainer>
                </Content>
              </AppNotification>
            )
          })}
      </Container>
    )
  }
}

export default injectIntl(AppNotifications)
