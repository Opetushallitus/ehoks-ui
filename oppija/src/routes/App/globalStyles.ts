import { createGlobalStyle } from "styled"
import { GlobalStyles as GlobalStylesBase } from "globalStyles"

export const GlobalStyles = createGlobalStyle`${GlobalStylesBase}`
