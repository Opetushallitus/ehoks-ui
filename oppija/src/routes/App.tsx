import { Routes, Route } from "react-router"
import { ThemeWrapper } from "components/ThemeWrapper"
import {
  cleanLocaleParam,
  isLocaleStored,
  parseLocaleParam,
  readLocaleFromDomain,
  readLocaleFromSessionStorage,
  setDocumentLocale
} from "localeUtils"
import { inject, observer } from "mobx-react"
import React, { useEffect } from "react"
import { IntlProvider } from "react-intl"
import { Ammatillinentutkinto } from "routes/Ammatillinentutkinto"
import { AppFooter } from "routes/App/AppFooter"
import { AppHeader } from "routes/App/AppHeader"
import AppNotifications from "routes/App/AppNotifications"
import { GlobalStyles } from "routes/App/globalStyles"
import { Etusivu } from "routes/Etusivu"
import { Henkilokohtaistaminen } from "routes/Henkilokohtaistaminen"
import { Suunnittelu } from "routes/Suunnittelu"
import { IRootStore } from "stores/RootStore"
import { Locale } from "stores/TranslationStore"
import styled from "styled"
import { Redirect } from "components/Redirect"

const Container = styled("div")`
  margin: 0;
  ${(props) =>
    props.theme.typography
      .body}/* Needed for fixing apply-raamit.css injection */
`

const Main = styled("main")``

const ModalContainer = styled("div")``

const MainApp = () => (
  <Container>
    <AppHeader />
    <AppNotifications />
    <Main id="main" role="main">
      <Routes>
        <Route index element={<Etusivu />} />
        <Route path="suunnittelu/*" element={<Suunnittelu />} />
        <Route
          path="henkilokohtaistaminen"
          element={<Henkilokohtaistaminen />}
        />
        <Route path="ammatillinentutkinto" element={<Ammatillinentutkinto />} />
      </Routes>
    </Main>
    <AppFooter />
    <GlobalStyles />
    <ModalContainer id="modal-root" />
  </Container>
)

export interface AppProps {
  store?: IRootStore
}

export const App = inject("store")(
  observer((props: AppProps) => {
    const { session, translations } = props.store!
    useEffect(() => {
      const localeParam = parseLocaleParam(window.location.search)
      if (localeParam) {
        translations.setActiveLocale(localeParam)
        cleanLocaleParam()
      } else {
        const locale = isLocaleStored()
          ? readLocaleFromSessionStorage()
          : readLocaleFromDomain()
        translations.setActiveLocale(locale)
      }

      // load user session info from backend
      session.checkSession()
    }, [session, translations])

    const activeLocale = translations.activeLocale
    setDocumentLocale(activeLocale)
    const localisedMessages = translations.messages[activeLocale]
    const messages =
      activeLocale === Locale.FI
        ? localisedMessages
        : // use finnish translations as fallback, merge provided translations
          { ...translations.messages.fi, ...localisedMessages }
    return (
      <ThemeWrapper>
        <IntlProvider
          defaultLocale={Locale.FI}
          locale={activeLocale}
          messages={messages}
          textComponent={React.Fragment}
        >
          <Routes>
            <Route path="/ehoks/*" element={<MainApp />} />
            <Route path="/" element={<Redirect to="/ehoks/" />} />
          </Routes>
        </IntlProvider>
      </ThemeWrapper>
    )
  })
)
