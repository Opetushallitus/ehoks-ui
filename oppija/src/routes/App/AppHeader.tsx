import { Link } from "react-router-dom"
import { getActiveDomain } from "localeUtils"
import { inject, observer } from "mobx-react"
import React, { useState } from "react"
import { MdMenu } from "react-icons/md"
import { useIntl, FormattedMessage } from "react-intl"
import { MobileMenu } from "routes/App/MobileMenu"
import { IRootStore } from "stores/RootStore"
import { Locale } from "stores/TranslationStore"
import styled from "styled"
import ehoksLogo from "./ehoks_logo.png"
import { FeedbackReminder } from "./FeedbackReminder"
import { LinkButton } from "components/Button"

interface TopLinkProps {
  active?: boolean
}

const HeaderContainer = styled("header")`
  width: 100%;
  color: #fff;
  background-color: ${(props) => props.theme.colors.header.background};
  line-height: 16px;
  font-size: 18px;
`

const TopLinksContainer = styled("div")`
  width: 100%;
  background-color: #254905;
  font-size: 16px;
  font-weight: 600;
  padding-left: 20px;

  @media screen and (max-width: ${(props) =>
      props.theme.breakpoints.Tablet}px) {
    display: none;
  }

  @media screen and (max-width: ${(props) =>
      props.theme.breakpoints.Desktop}px) {
    padding-left: 0;
  }
`

const TopLinks = styled("nav")`
  line-height: 20px;
  max-width: ${(props) => props.theme.maxWidth}px;
  margin: 0 auto;
`

const SkipToMain = styled("a")`
  left: -999px;
  position: absolute;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
  z-index: -999;

  &:focus,
  &:active {
    position: unset;
    color: #fff;
    width: auto;
    height: auto;
    padding: 5px 20px 5px 20px;
    line-height: 20px;
    z-index: 999;
  }
`

const TopLink = styled("a")<TopLinkProps>`
  display: inline-block;
  padding: 5px 20px 5px 20px;
  text-decoration: none;
  color: #fff;
  line-height: 20px;
  background: ${(props) =>
    props.active ? props.theme.colors.header.background : "transparent"};

  &:focus {
    outline: 2px solid ${(props) => props.theme.colors.green300};
  }
`

const TitleContainer = styled("div")`
  max-width: ${(props) => props.theme.maxWidth}px;
  margin: 0 auto;
  display: flex;
  align-items: center;
`

const MobileMenuToggle = styled("div")`
  display: none;
  @media screen and (max-width: ${(props) =>
      props.theme.breakpoints.Desktop}px) {
    display: block;
    margin: 0 20px;

    h3 {
      margin: -5px 0 0 0;
      font-size: 12px;
      font-weight: 300;
      text-transform: uppercase;
      text-align: center;
    }
  }
`

const Logo = styled("img")`
  height: 64px;
  margin: 4px 0 4px 40px;

  @media screen and (max-width: ${(props) =>
      props.theme.breakpoints.Desktop}px) {
    margin: 4px 0 4px 20px;
  }
`

const LogoutContainer = styled("div")`
  display: block;
  margin: 0 40px 0 0;
  font-weight: 300;

  @media screen and (max-width: ${(props) =>
      props.theme.breakpoints.Tablet}px) {
    display: none;
  }

  @media screen and (max-width: ${(props) =>
      props.theme.breakpoints.Desktop}px) {
    margin-right: 20px;
  }
`

const User = styled("div")`
  font-weight: 400;
  margin-bottom: 5px;
`

const LogoutLink = styled(Link)`
  color: #fff;
`

const Flex = styled("div")`
  flex: 1;
`

interface LanguageSelectorProps {
  loggedIn: boolean
}
const LanguageSelector = styled("div")<LanguageSelectorProps>`
  margin: ${(props) => (props.loggedIn ? "0 40px 0 0" : "0 80px 0 0")};
  a {
    margin-right: 10px;
    font-size: 15px;
    cursor: pointer;
  }

  @media screen and (max-width: ${(props) =>
      props.theme.breakpoints.Tablet}px) {
    display: none;
  }
`

interface AppHeaderProps {
  store?: IRootStore
}

interface AppHeaderState {
  showMenu: boolean
}

export const AppHeader = inject("store")(
  observer((props: AppHeaderProps) => {
    const [state, setState] = useState<AppHeaderState>({
      showMenu: false
    })

    const logoutOppija = async (event: React.MouseEvent) => {
      event.preventDefault()
      const rootStore = props.store
      // featureflags.casOppija
      const casOppijaFeatureFlag = true
      if (casOppijaFeatureFlag) {
        try {
          // Ends ehoks session, doesn't end cas session
          await rootStore!.session.logoutOppija()
        } finally {
          // Ends cas session
          window.location.href =
            rootStore!.translations.activeLocale === Locale.FI
              ? rootStore!.environment.casOppijaLogoutUrlFi
              : rootStore!.environment.casOppijaLogoutUrlSv
        }
      } else {
        rootStore!.session.logoutOppija()
      }
    }

    const changeLocale = (locale: Locale) => (event: React.MouseEvent) => {
      event.preventDefault()
      props.store!.translations.setActiveLocale(locale)
    }

    const toggleMenu = () => {
      setState({ ...state, showMenu: !state.showMenu })
    }

    const showFeedbackModal = () => {
      props.store!.notifications.makeFeedbackModalVisible()
    }

    const { store } = props
    const intl = useIntl()
    const { session } = store!
    const { user, isLoggedIn } = session!
    const { activeLocale } = store!.translations
    const hasUnansweredFeedbackLinks =
      store!.notifications.hasUnanswaredFeedbackLinks
    const activeDomain = getActiveDomain()
    return (
      <HeaderContainer>
        <TopLinksContainer>
          <TopLinks>
            <SkipToMain href="#main">
              <FormattedMessage
                id="header.skipToMainLink"
                defaultMessage="Siirry pääsisältöön"
              />
            </SkipToMain>
            <TopLink href={`https://${activeDomain}/`}>
              <FormattedMessage
                id="header.opintopolkuLink"
                defaultMessage="Opintopolku.fi"
              />
            </TopLink>
            <TopLink href={`https://${activeDomain}/oma-opintopolku/`}>
              <FormattedMessage
                id="header.omaOpintopolkuLink"
                defaultMessage="Oma Opintopolku"
              />
            </TopLink>
            <TopLink href={`https://eperusteet.${activeDomain}/`}>
              <FormattedMessage
                id="header.ePerusteetLink"
                defaultMessage="ePerusteet"
              />
            </TopLink>
            <TopLink active={true} href="/">
              <FormattedMessage id="header.ehoksLink" defaultMessage="eHOKS" />
            </TopLink>
          </TopLinks>
        </TopLinksContainer>
        <TitleContainer>
          <MobileMenuToggle onClick={toggleMenu}>
            <MdMenu size="40" />
            <h3>
              <FormattedMessage
                id="header.mobiilivalikkoTitle"
                defaultMessage="Valikko"
              />
            </h3>
          </MobileMenuToggle>

          <Logo
            src={ehoksLogo}
            alt={intl.formatMessage({
              id: "header.ehoksLogoLabel"
            })}
          />

          <FeedbackReminder
            hasFeedbackLinks={hasUnansweredFeedbackLinks}
            onClick={showFeedbackModal}
          />
          <Flex />

          <LanguageSelector loggedIn={isLoggedIn}>
            {activeLocale === Locale.FI ? (
              <LinkButton onClick={changeLocale(Locale.SV)}>
                <FormattedMessage
                  id="header.swedishLocaleLink"
                  defaultMessage="På svenska"
                />
              </LinkButton>
            ) : (
              <LinkButton onClick={changeLocale(Locale.FI)}>
                <FormattedMessage
                  id="header.finnishLocaleLink"
                  defaultMessage="Suomeksi"
                />
              </LinkButton>
            )}
          </LanguageSelector>

          {isLoggedIn && user && (
            <LogoutContainer>
              <User>
                {user.usingValtuudet ? (
                  <FormattedMessage
                    id="header.valtuudet"
                    defaultMessage="(Valtuudet)"
                  />
                ) : null}
                {user.firstName} {user.surname}
              </User>
              <LogoutLink to="" onClick={logoutOppija}>
                <FormattedMessage
                  id="header.kirjauduUlosLink"
                  defaultMessage="Kirjaudu ulos"
                />
              </LogoutLink>
            </LogoutContainer>
          )}
        </TitleContainer>
        {state.showMenu && (
          <MobileMenu
            activeLocale={activeLocale}
            changeLocale={changeLocale}
            toggleMenu={toggleMenu}
            logout={logoutOppija}
            session={session!}
          />
        )}
      </HeaderContainer>
    )
  })
)
