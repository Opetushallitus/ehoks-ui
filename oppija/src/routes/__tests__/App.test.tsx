import {
  createHistory,
  createMemorySource,
  LocationProvider
} from "@reach/router"
import { apiUrl } from "config"
import { createEnvironment } from "createEnvironment"
import { mockFetch } from "fetchUtils"
import "jest-dom/extend-expect"
import "jest-styled-components"
import { Provider } from "mobx-react"
import React from "react"
import { cleanup, render } from "react-testing-library"
import { RootStore } from "stores/RootStore"
import { App } from "../App"

function renderWithRouter(
  ui: any,
  { route = "/", history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...render(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

afterEach(cleanup)

it("App renders correctly", async () => {
  // pass fetch utils to root store using store's environment context
  const store = RootStore.create(
    {},
    createEnvironment(mockFetch(apiUrl), apiUrl, "")
  )
  // fetch default translations
  await store.translations.fetchLocales()

  const { container } = renderWithRouter(
    <Provider store={store}>
      <App />
    </Provider>,
    { route: "/ehoks" }
  )
  expect(container).toMatchSnapshot()
})
