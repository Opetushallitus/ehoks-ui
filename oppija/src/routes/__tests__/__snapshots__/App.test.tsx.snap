// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
.c10 {
  max-width: 1440px;
  margin: 0 auto;
}

.c21 {
  height: 50px;
}

.c21:first-of-type {
  margin-right: 20px;
}

.c17 {
  margin-left: 20px;
  width: 30px;
}

.c11 {
  width: 100%;
  background-color: #f8f8f8;
  border-top: 1px solid #c8cdcf;
}

.c12 {
  max-width: 1440px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 20px 50px 30px 40px;
}

.c13 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c18 {
  list-style: none;
  margin: 10px 0 0 0;
  padding-left: 0;
}

.c18 li {
  line-height: 30px;
  color: #4a4a4a;
}

.c19 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: transparent;
}

.c20 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c22 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex: 4;
  -ms-flex: 4;
  flex: 4;
}

.c22 p {
  font-size: 14px;
  line-height: 24px;
  text-align: right;
  color: #626769;
  margin-bottom: 30px;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c15 {
  width: 200px;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c24 a {
  margin-right: 10px;
  font-size: 13px;
  cursor: pointer;
}

.c25 {
  color: #cfd5d7;
}

.c16 {
  display: none;
}

.c26 {
  display: block;
}

.c1 {
  width: 100%;
  color: #fff;
  background-color: #149ecb;
  line-height: 16px;
  font-size: 18px;
  font-weight: 600;
}

.c2 {
  width: 100%;
  background-color: #06526b;
  font-size: 16px;
  padding-left: 20px;
}

.c3 {
  line-height: 20px;
  max-width: 1440px;
  margin: 0 auto;
}

.c4 {
  left: -999px;
  position: absolute;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
  z-index: -999;
}

.c4:focus,
.c4:active {
  position: unset;
  color: #fff;
  width: auto;
  height: auto;
  padding: 5px 20px 5px 20px;
  line-height: 20px;
  z-index: 999;
}

.c5 {
  display: inline-block;
  padding: 5px 20px 5px 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  line-height: 20px;
  background: transparent;
}

.c5:focus {
  outline: 2px solid #1976d2;
}

.c6 {
  display: inline-block;
  padding: 5px 20px 5px 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  line-height: 20px;
  background: #149ecb;
}

.c6:focus {
  outline: 2px solid #1976d2;
}

.c7 {
  max-width: 1440px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: none;
}

.c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 32px;
  line-height: 100%;
  font-weight: 400;
  margin: 25px 5px 25px 40px;
}

.c0 {
  margin: 0;
}

@media screen and (max-width:900px) {
  .c17 {
    margin-top: 15px;
    margin-left: 10px;
  }

  a:first-of-type .c17 {
    margin-left: 0;
  }
}

@media screen and (max-width:900px) {
  .c11 {
    background-color: #fff;
  }
}

@media screen and (max-width:900px) {
  .c12 {
    display: block;
    padding: 0;
  }
}

@media screen and (max-width:900px) {
  .c13 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 20px 50px 30px 20px;
  }
}

@media screen and (max-width:900px) {
  .c19 {
    background: #f8f8f8;
    border-top: 1px solid #979797;
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (max-width:900px) {
  .c20 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    margin: 20px 0;
  }
}

@media screen and (max-width:900px) {
  .c22 {
    padding: 0 20px;
  }

  .c22 p {
    margin-bottom: 0;
    text-align: left;
  }
}

@media screen and (max-width:900px) {
  .c15 {
    width: 150px;
  }
}

@media screen and (max-width:900px) {
  .c23 {
    display: none;
  }
}

@media screen and (max-width:900px) {
  .c16 {
    display: block;
  }
}

@media screen and (max-width:900px) {
  .c26 {
    display: none;
  }
}

@media screen and (max-width:720px) {
  .c2 {
    display: none;
  }
}

@media screen and (max-width:900px) {
  .c2 {
    padding-left: 0;
  }
}

@media screen and (max-width:900px) {
  .c8 {
    display: block;
    margin: 0 20px;
  }

  .c8 h3 {
    margin: -5px 0 0 0;
    font-size: 12px;
    font-weight: 300;
    text-transform: uppercase;
    text-align: center;
  }
}

@media screen and (max-width:900px) {
  .c9 {
    margin: 25px 5px 25px 20px;
  }
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <div
      className="c2"
    >
      <nav
        className="c3"
      >
        <a
          className="c4"
          href="#main"
        >
          Siirry pääsisältöön
        </a>
        <a
          className="c5"
          href="https://opintopolku.fi/"
        >
          Opintopolku.fi
        </a>
        <a
          className="c5"
          href="https://opintopolku.fi/oma-opintopolku/"
        >
          Oma Opintopolku
        </a>
        <a
          className="c5"
          href="https://eperusteet.opintopolku.fi/"
        >
          ePerusteet
        </a>
        <a
          className="c6"
          href="/"
        >
          eHOKS
        </a>
      </nav>
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="40"
          size="40"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="40"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <h3>
          Valikko
        </h3>
      </div>
      <div
        className="c9"
      >
        eHOKS
      </div>
    </div>
  </header>
  <div
    className="c10"
  />
  <main
    className=""
    id="main"
    role="main"
  />
  <footer
    className="c11"
  >
    <div
      className="c12"
    >
      <div
        className="c13"
      >
        <div
          className="c14"
        >
          <img
            alt="Opintopolku logo"
            className="c15"
            src=""
          />
          <div
            className="c16"
          >
            <a>
              <img
                alt="Twitter"
                className="c17"
                src=""
              />
            </a>
            <a>
              <img
                alt="Facebook"
                className="c17"
                src=""
              />
            </a>
            <a>
              <img
                alt="Instagram"
                className="c17"
                src=""
              />
            </a>
          </div>
        </div>
        <ul
          className="c18"
        >
          <li>
            ePerusteet
          </li>
          <li>
            Mikä on opintopolku
          </li>
          <li>
            Tietoturvaseloste
          </li>
          <li>
            Anna palautetta
          </li>
        </ul>
      </div>
      <div
        className="c19"
      >
        <div
          className="c20"
        >
          <img
            alt="Opetus- ja kulttuuriministeriö logo"
            className="c21"
            src=""
          />
          <img
            alt="Opetushallitus logo"
            className="c21"
            src=""
          />
        </div>
        <div
          className="c22"
        >
          <p>
            Koulutuksen järjestäjät ja korkeakoulut ylläpitävät tietoja
              koulutuksistaan Opintopolku.fi-palvelussa. Tietojen oikeellisuuden
              voit tarkistaa kyseisestä oppilaitoksesta tai korkeakoulusta.
          </p>
          <div
            className="c23"
          >
            <div
              className="c24"
            >
              <a
                onClick={[Function]}
                role="button"
              >
                Suomeksi
              </a>
              <a
                onClick={[Function]}
                role="button"
              >
                På svenska
              </a>
            </div>
            <div
              className="c25"
            >
              |
            </div>
            <div
              className="c26"
            >
              <a>
                <img
                  alt="Twitter"
                  className="c17"
                  src=""
                />
              </a>
              <a>
                <img
                  alt="Facebook"
                  className="c17"
                  src=""
                />
              </a>
              <a>
                <img
                  alt="Instagram"
                  className="c17"
                  src=""
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
